test/unit/sharedlib_test.go

```diff
@@ -341,6 +360,63 @@ func (s shellScript) write(t TestingT, src string) string {
 	return p
 }
 
+type prefetcher interface {
+	prefetch(t TestingT)
+}
+
+type fnPrefetcher func(t TestingT)
+
+func (f fnPrefetcher) prefetch(t TestingT) {
+	f(t)
+}
+
+func goRunHelpPrefetcher(tool string) prefetcher {
+	return fnPrefetcher(func(t TestingT) {
+		c := exec.Command("go", "run", tool, "--help")
```

Comment:
why you're calling help here! Could you please clarify better the purpose of the pre-fetcher

Reply:
I added that prefetcher code to the tests, so to make sure the tools are already downloaded and compiled. Without it, the tests might fetch those tools during runtime, and that was causing tests to have unexpected output: https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative_hack/238/unit-tests_hack_main/1582291191687286784

Maybe I'll add some comments to the source code to make it clear.